### PR TITLE
Course & Chachkies 

### PR DESCRIPTION
**Pull request by chr0n1x**
Bucket touched courses, experimental badge, progress wizard.

## Description

- Add a progress wizard for course templates
- `ExperimentalBadge` because this is prone to...my bad code
- Courses can be "completed" now and `NextPrev` navigation stops based on dir

## Related Issue

N/A

## Motivation and Context

Ethos needs a proof-of-concept to show around.

## How Has This Been Tested?

Locally - ran app & ran `yarn build`

#### Course Changes
Screen Shot 2021-04-22 at 3 03 25 PM

#### New Course Catalog Template

Screen Shot 2021-04-22 at 3 09 26 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/components/ExperimentalBadge.jsx [hunk 1]
```diff
@@ -10,14 +10,18 @@
...
+import { React } from "react"
+import "@spectrum-css/label"
 
 const ExperimentalBadge = ({ inline }) => (
-  <Fragment>
-    <Badge content={"EXPERIMENTAL"} />
-    { !inline && <br /> }
-  </Fragment>
+  <span
+    class="spectrum-Label spectrum-Label--red"
```
Comment (chr0n1x) [inline src/components/ExperimentalBadge.jsx:18]: 🥳

File: src/components/ExperimentalBadge.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ *  OF ANY KIND, either express or implied. See the License for the specific language
+ *  governing permissions and limitations under the License.
+ */
+
+import { React } from "react"
+import "@spectrum-css/label"
+
+const ExperimentalBadge = ({ inline }) => (
+  <span
+    class="spectrum-Label spectrum-Label--red"
```
Comment (macdonst) [inline src/components/ExperimentalBadge.jsx:18]: @chr0n1x sorry can you change `class` to `className`. That is what we are supposed to use in React.

File: src/templates/course-catalog.js [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+  flat = flat.flat()
+  return flat.filter(
+    (page, index) => page.path && page.path !== flat[index + 1]?.path
+  )
+}
+
+const courseButton = (page) => (
+  <ActionButton minWidth={300} margin={4}>
+    <a href={page.path}>{page.title}</a>
```
Comment (macdonst) [inline src/templates/course-catalog.js:83]: Alright, I see why there is different behaviour. You don't want to use raw url as when the project is compile with the path prefix this href will probably be invalid. Instead use:

```
<a onClick={() => navigate(page.path)}>{page.title}</a>
```

Of course, you'd have to import it from `gatsby`

```
import { graphql, navigate } from "gatsby"
```

**Review by macdonst — CHANGES_REQUESTED** [review]
Please make the changes I indicated and let me know so I can merge. It should resolve the problems I was running into that could crash the page.

**Review by macdonst — APPROVED** [review]

**Pull request merged**